server/controllers/requestController.js

```diff
@@ -61,15 +66,18 @@ exports.updateRequest = async (req, res, next) => {
     if (!ObjectId.isValid(req.params.id)) {
       return next(createError(400, "Invalid Request ID!"));
     }
-    const request = await Request.findById(req.params.id);
+    const request = await Request.findById(req.params.id).populate([
+      { path: "sitterId", model: "User", populate: { path: "profile" } },
+      { path: "ownerId", model: "User", populate: { path: "profile" } },
+    ]);
 
     // Check if request exists
     if (!request) {
       return next(createError(404, "Update Failed! Request does not exist!"));
     }
 
     // Only the sitter can accept/decline request
-    if (request.sitterId.toString() !== req.user.id) {
+    if (request.sitterId.id.toString() !== req.user.id) {
```

Comment:
I think this should be ._id
```suggestion
    if (request.sitterId._id.toString() !== req.user.id) {
```

Reply:
Changed to ._id.